gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/matching/single/cypher/operators/expand/ExpandEmbeddingsIteration.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of Gradoop.
+ *
+ * Gradoop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Gradoop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Gradoop. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.expand;
+
+import org.apache.flink.api.common.operators.base.JoinOperatorBase;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.operators.IterativeDataSet;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.ExpandDirection;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.common.pojos.Embedding;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.expand.functions.FilterPreviousExpandEmbedding;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.expand.functions.MergeExpandEmbeddings;
+
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.operators.expand.tuples.ExpandEmbedding;
+
+import java.util.List;
+
+/**
+ * Expands an vertex along the edges. The number of hops can be specified via upper and lower bound
+ * The input embedding is appended by 2 Entries, the first one represents the path (edge, vertex,
+ * edge, vertex, ..., edge), the second one the end vertex
+ *
+ * Iteration is done with {@code BulkIteration}
+ */
+public class ExpandEmbeddingsIteration extends ExpandEmbeddings {
```

Comment:
let's call this one 'ExpandEmbeddingsBulk' and the other one 'ExpandEmbeddingsForLoop'